### PR TITLE
Build and publish docker images

### PR DESCRIPTION
**Pull request by merlimat**
### Motivation

Build the docker images during the CI build and push them to DockerHub when a final tag is created.

### CODE REVIEW

**Comment by merlimat** [comment]
Tested pushing from my repo to personal docker hub account: 

https://hub.docker.com/u/merlimat/ 

https://travis-ci.org/merlimat/pulsar/builds/254662796

**Comment by merlimat** [comment]
@saandrews By default it will publish on:
 * `apachepulsar/pulsar`
 * `apachepulsar/pulsar-dashboard`
 * `apachepulsar/pulsar-grafana`

The `DOCKER_ORG` variable controls the org account. I have created the `apachepulsar` already and can add people to control that. I just need the Docker Hub account names.

File: docker/publish.sh [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+# under the License.
+#
+
+ROOT_DIR=$(git rev-parse --show-toplevel)
+cd $ROOT_DIR/docker
+
+# We should only publish images that are made from official and approved releases
+# and thus ignore all the release candidates that are just tagged during the release
+# process
+if [[ ${TRAVIS_TAG} == *"candidate"* || ${TRAVIS_TAG} == *"rc"* ]]; then
```
Comment (rdhabalia) [inline docker/publish.sh]: where are we setting `TRAVIS_TAG`, was it set by travis when we were using it?

Comment (merlimat) [reply docker/publish.sh]: Yes, this was working on Travis. It hasn't been ported to Jenkins yet. I did manually published the last release.

Comment (merlimat) [reply docker/publish.sh]: I mean, the script still works. It's just that we need to chose when to trigger it. Eg: right now we need to cut the release, get it approved and then publish it, so it's a different workflow from the Travis based release where everything was done in 1 shot.

**Comment by rdhabalia** [comment]
> right now we need to cut the release, get it approved and then publish i

I see, so do we have a separate jenkins job other than [master-build](https://builds.apache.org/job/pulsar-master/) to publish docker-build.?

**Comment by merlimat** [comment]
> I see, so do we have a separate jenkins job other than master-build to publish docker-build.?

We don't but we should. And it should be a job to be manually triggered on releases

**Pull request merged**